File: app/src/main/java/kashyap/genovatest/MyApplication.java
package kashyap.genovatest;

import android.app.Application;
import android.content.Context;
import android.os.Build;
import android.text.TextUtils;
import android.util.Log;



import java.security.KeyManagementException;
import java.security.NoSuchAlgorithmException;

import javax.net.ssl.SSLContext;

import kashyap.genovatest.connect.ConnectivityReceiver;


public class MyApplication extends Application {
    Context context;
    public static final String TAG = MyApplication.class.getSimpleName();
    private static MyApplication mInstance;
    @Override
    public void onCreate() {
        super.onCreate();
        context = this;
        mInstance = this;

    }
    public static synchronized MyApplication getInstance() {
        return mInstance;
    }

    @Override
    protected void attachBaseContext(Context context) {
        super.attachBaseContext(context);
    }


    public void setConnectivityListener(ConnectivityReceiver.ConnectivityReceiverListener listener) {
        ConnectivityReceiver.connectivityReceiverListener = listener;
    }
}

File: app/src/main/java/kashyap/genovatest/MyApplication.kt
package kashyap.genovatest

import android.app.Application
import android.content.Context
import kashyap.genovatest.connect.ConnectivityReceiver
import kashyap.genovatest.connect.ConnectivityReceiver.ConnectivityReceiverListener

class MyApplication : Application() {
    var context: Context? = null
    override fun onCreate() {
        super.onCreate()
        context = this
        instance = this
    }

    override fun attachBaseContext(context: Context) {
        super.attachBaseContext(context)
    }

    fun setConnectivityListener(listener: ConnectivityReceiverListener?) {
        ConnectivityReceiver.connectivityReceiverListener = listener
    }

    companion object {
        val TAG = MyApplication::class.java.simpleName

        @get:Synchronized
        var instance: MyApplication? = null
            private set
    }
}